include/deal.II/grid/grid_tools.h

```diff
@@ -556,6 +556,137 @@ namespace GridTools
+                                  const std::set<types::material_id> &materials);
+
+  /**
+   * Extract and return all active cells around a subdomain defined by a
+   * material id.
+   *
+   * @tparam Container A type that satisfies the requirements of a mesh
+   * container (see @ref GlossMeshAsAContainer).
+   * @param[in] container A mesh container (i.e. objects of type Triangulation,
+   * DoFHandler, or hp::DoFHandler).
+   * @param[in] materials The material id that defines the subdomain that is
+   * encapsulated by the cell layer that is to be extracted.
+   * @return The list of cells that comprise the halo layer around the
+   * specified material
+   *
+   * @note The returned vector of cells may include both locally owned and
+   * ghost cells, but will not include artificial cells.
+   *
+   * @author Jean-Paul Pelteret, Denis Davydov, 2015
+   */
+  template <class Container>
+  std::vector<typename Container::active_cell_iterator>
+  compute_active_cell_halo_layer (const Container          &container,
+                                  const types::material_id &material);
+
+  /**
+   * Extract and return all active cells around a subdomain defined by a
+   * collection of active FE indices.
+   *
+   * @param[in] dof_handler A container of type hp::DoFHandler.
+   * @param[in] active_fe_indices A collection of active FE indices that define
+   * the subdomain that is encapsulated by the cell layer that is to be extracted.
+   * @return The list of cells that comprise the halo layer around the
+   * specified set of FE indices
+   *
+   * @note The returned vector of cells may include both locally owned and
+   * ghost cells, but will not include artificial cells.
+   *
+   * @author Jean-Paul Pelteret, Denis Davydov, 2015
+   */
+  template <int dim, int spacedim>
+  std::vector<typename hp::DoFHandler<dim,spacedim>::active_cell_iterator>
+  compute_active_cell_halo_layer (const hp::DoFHandler<dim,spacedim> &dof_handler,
+                                  const std::set<unsigned int>       &active_fe_indices);
+
+  /**
+   * Extract and return all active cells around a subdomain defined by an
+   * active FE index.
+   *
+   * @param[in] dof_handler A container of type hp::DoFHandler
+   * @param[in] active_fe_index The active FE index that defines the subdomain
+   * that is encapsulated by the cell layer that is to be extracted.
+   * @return The list of cells that comprise the halo layer around the
+   * specified FE index
+   *
+   * @note The returned vector of cells may include both locally owned and
+   * ghost cells, but will not include artificial cells.
+   *
+   *
+   * @author Jean-Paul Pelteret, Denis Davydov, 2015
+   */
+  template <int dim, int spacedim>
+  std::vector<typename hp::DoFHandler<dim,spacedim>::active_cell_iterator>
+  compute_active_cell_halo_layer (const hp::DoFHandler<dim,spacedim> &dof_handler,
+                                  const unsigned int                 &active_fe_index);
```

Comment:
I'm not totally convinced that we want four overloads for this one function that all essentially have just a one-line implementation. I think if we introduce these four overloads, then we should probably do the same for other argument types as well. But even with just the current state you get into a bit of trouble -- for example, which overload does this call:
```
  compute_active_cell_halo_layer (hp_dof_handler, 1);
```
Here, `1` has type `int`. Is it going to be converted to `unsigned int` (and then be interpreted as the active FE index) or to `types::material_id`? I foresee people getting into trouble this way.

I think my preference would be to remove these four overloads and instead extend the documentation of the original function by a section that shows some examples, and these examples would simply be the implementations of the four overloads you have here.